formset.py

```diff
@@ -753,84 +830,21 @@ def main():
         print("-m{0}x{1}".format(total_cpus // cpus, cpus))
         exit()
 
-    # Presumably increasing MaxTermSize requires increasing WorkSpace, too.
-    sp.workspace = max(sp.workspace, sp.maxtermsize * 250)
+    sp, x = sp.scale(memory, human_readable=args.human_readable)
 
-    # Optimize the memory usage by bisection.
-    max_iteration = 50
+    # Final memory usage we've found.
+    memory_usage = sp.calc()
 
-    sp0 = sp.copy()
-
-    def f(x):
-        # type: (float) -> Tuple[int, Setup]
-        # Hopefully monochrome increasing.
-        sp = sp0.copy()
-        sp.smallsize = int(sp.smallsize * x)
-        sp.largesize = int(sp.largesize * x)
-        sp.termsinsmall = int(sp.termsinsmall * x)
-        sp.scratchsize = int(sp.scratchsize * x)
-        m = sp.calc()
-        if args.human_readable:
-            m = round_human_readable(m, True, False)
-        return (-(memory - m), sp)
-
-    x1 = 1.0
-    x2 = None  # type: Optional[float]
-    y1 = f(x1)[0]
-    y2 = None  # type: Optional[int]
-    for _i in range(max_iteration):
-        if x2 is None:
-            if y1 < 0:
-                x = x1 * 2.0
-                y = f(x)[0]
-                if y > 0:
-                    x2 = x
-                    y2 = y
-                else:
-                    x1 = x
-                    y1 = y
-            else:
-                x = x1 * 0.5
-                y = f(x)[0]
-                if y < 0:
-                    x2 = x1
-                    y2 = y1
-                    x1 = x
-                    y1 = y
-                else:
-                    x1 = x
-                    y1 = y
-        else:
-            x = (x1 + x2) * 0.5
-            y = f(x)[0]
-            if y < 0:
-                x1 = x
-                y1 = y
-            else:
-                x2 = x
-                y2 = y
-        if x2 is not None:
-            # sanity checks
-            if y2 is None:
-                raise AssertionError()
-            if x1 >= x2:
-                raise AssertionError()
-            if y1 >= y2:
-                raise AssertionError()
-
-    if x2 is None:
-        if x1 < 1.0e-12:
-            x1 = 0
+    if x < 1e-12:
         parser.exit(
             -1,
-            ("failed to find parameters: memory({0}) = {1} " "bytes shortage").format(
-                x1, y1
+            ("failed to find parameters: memory({0}) = {1} bytes shortage\n").format(
```

Comment:
The new line `\n` at the end seems not needed. Actually, it doesn't change the error message, so I like not adding it.

Reply:
It does change it; that newline is not printed automatically, and running `./formset.py -p 0` messes up the bash prompt without it.

Reply:
Oh, sorry. You are right. In the fish shell it looks same, but there is a tiny mark indicating new line is missing, which I missed.